### ISSUE TITLE
pod2html having problems with B<< foo->{bar} >>

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10516]
Migrated from [rt.perl.org#76752](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=76752) (status was 'resolved')

Searchable as RT76752$

**Comment by p5pRT** [comment]
### From jpl@research.att.com

#### Created by jpl@research.att.com



File t\.pod contains

&nbsp;    =head2 title

&nbsp;    B\<\< foo\->\{bar\} >>

If I run

&nbsp;    /honk/perl5\.12\.1/bin/pod2html t\.pod > t\.html

\(to get the 5\.12\.1 version\)\, the relevant part of t\.html
looks like

&nbsp;    \<h2>\<a name="title">title\</a>\</h2>
&nbsp;    \<p>\<strong>foo\-\</strong>\{bar\} &gt;&gt;\</p>

So the > in foo\-> is closing the B\<\<\, and the proper
closing pattern >> is getting escaped verbatim\.
This is true of every release of pod2html I can find\.
I must be doing something wrong\, this cannot have gone
unnoticed for years\.  But it sure looks like I'm using
the B\<\< \.\.\. >> escape exactly as perldoc perlpod specifies\,
with a blank after the \<\< and before the >>\.  \-\- jpl

PS&#8203;: I see this report is using perl5\.8\.8 as appears on
the machine where I send mail\.  Same problem there\.

<details><summary>Perl Info</summary>

```

Flags:
    category=utilities
    severity=medium

This perlbug was built using Perl v5.8.8 in the Red Hat build system.
It is being executed now by Perl v5.8.8 - Mon Jun  7 21:25:07 EDT 2010.

Site configuration information for perl v5.8.8:

Configured by Red Hat, Inc. at Mon Jun  7 21:25:07 EDT 2010.

Summary of my perl5 (revision 5 version 8 subversion 8) configuration:
  Platform:
    osname=linux, osvers=2.6.18-53.el5, archname=x86_64-linux-thread-multi
    uname='linux builder10.centos.org 2.6.18-53.el5 #1 smp mon nov 12 02:14:55 est 2007 x86_64 x86_64 x86_64 gnulinux '
    config_args='-des -Doptimize=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -Dversion=5.8.8 -Dmyhostname=localhost -Dperladmin=root@localhost -Dcc=gcc -Dcf_by=Red Hat, Inc. -Dinstallprefix=/usr -Dprefix=/usr -Dlibpth=/usr/local/lib64 /lib64 /usr/lib64 -Dprivlib=/usr/lib/perl5/5.8.8 -Dsitelib=/usr/lib/perl5/site_perl/5.8.8 -Dvendorlib=/usr/lib/perl5/vendor_perl/5.8.8 -Darchlib=/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi -Dsitearch=/usr/lib64/perl5/site_perl/5.8.8/x86_64-linux-thread-multi -Dvendorarch=/usr/lib64/perl5/vendor_perl/5.8.8/x86_64-linux-thread-multi -Darchname=x86_64-linux-thread-multi -Dvendorprefix=/usr -Dsiteprefix=/usr -Duseshrplib -Dusethreads -Duseithreads -Duselargefiles -Dd_dosuid -Dd_semctl_semun -Di_db -Ui_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Duseperlio -Dinstallusrbinperl=n -Ubincompat5005 -Uversiononly -Dpager=/usr/bin/less -isr -Dd_gethostent_r_proto -Ud_end
 hostent_r_proto -Ud_sethostent_r_proto -Ud_endprotoent_r_proto -Ud_setprotoent_r_proto -Ud_endservent_r_proto -Ud_setservent_r_proto -Dinc_version_list=5.8.7 5.8.6 5.8.5 -Dscriptdir=/usr/bin'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=define uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='4.1.2 20080704 (Red Hat 4.1.2-48)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =''
    libpth=/usr/local/lib64 /lib64 /usr/lib64
    libs=-lresolv -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic'

Locally applied patches:
    


@INC for perl v5.8.8:
    /usr/lib64/perl5/site_perl/5.8.8/x86_64-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.8
    /usr/lib/perl5/site_perl
    /usr/lib64/perl5/vendor_perl/5.8.8/x86_64-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.8.8
    /usr/lib/perl5/vendor_perl
    /usr/lib64/perl5/5.8.8/x86_64-linux-thread-multi
    /usr/lib/perl5/5.8.8
    .


Environment for perl v5.8.8:
    HOME=/home/jpl
    LANG=C
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/jpl/x86_64bin:/home/jpl/bin:/bin:/usr/bin:/usr/local/bin:/etc:/usr/sbin:/usr/etc:/usr/common/bin:/usr/games:.
    PERL_BADLANG (unset)
    SHELL=/bin/ksh

```
</details>

**Comment by p5pRT** [comment]
### From david@kineticode.com

Time to switch pod2html to use Pod&#8203;::Simple&#8203;::HTMLLegacy?

&nbsp;  http&#8203;://www\.mail\-archive\.com/pod\-people@&#8203;perl\.org/msg01326\.html

Best\,

David

On Jul 27\, 2010\, at 7&#8203;:32 AM\, jpl@&#8203;research\.att\.com \(via RT\) wrote&#8203;:

> \# New Ticket Created by  jpl@&#8203;research\.att\.com 
> \# Please include the string&#8203;:  \[perl \#76752\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=76752 >
> 
> 
> 
> This is a bug report for perl from jpl@&#8203;research\.att\.com\,
> generated with the help of perlbug 1\.35 running under perl v5\.8\.8\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> File t\.pod contains
> 
>    =head2 title
> 
>    B\<\< foo\->\{bar\} >>
> 
> If I run
> 
>    /honk/perl5\.12\.1/bin/pod2html t\.pod > t\.html
> 
> \(to get the 5\.12\.1 version\)\, the relevant part of t\.html
> looks like
> 
>    \<h2>\<a name="title">title\</a>\</h2>
>    \<p>\<strong>foo\-\</strong>\{bar\} &gt;&gt;\</p>
> 
> So the > in foo\-> is closing the B\<\<\, and the proper
> closing pattern >> is getting escaped verbatim\.
> This is true of every release of pod2html I can find\.
> I must be doing something wrong\, this cannot have gone
> unnoticed for years\.  But it sure looks like I'm using
> the B\<\< \.\.\. >> escape exactly as perldoc perlpod specifies\,
> with a blank after the \<\< and before the >>\.  \-\- jpl
> 
> PS&#8203;: I see this report is using perl5\.8\.8 as appears on
> the machine where I send mail\.  Same problem there\.
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>    category=utilities
>    severity=medium
> \-\-\-
> This perlbug was built using Perl v5\.8\.8 in the Red Hat build system\.
> It is being executed now by Perl v5\.8\.8 \- Mon Jun  7 21&#8203;:25&#8203;:07 EDT 2010\.
> 
> Site configuration information for perl v5\.8\.8&#8203;:
> 
> Configured by Red Hat\, Inc\. at Mon Jun  7 21&#8203;:25&#8203;:07 EDT 2010\.
> 
> Summary of my perl5 \(revision 5 version 8 subversion 8\) configuration&#8203;:
>  Platform&#8203;:
>    osname=linux\, osvers=2\.6\.18\-53\.el5\, archname=x86\_64\-linux\-thread\-multi
>    uname='linux builder10\.centos\.org 2\.6\.18\-53\.el5 \#1 smp mon nov 12 02&#8203;:14&#8203;:55 est 2007 x86\_64 x86\_64 x86\_64 gnulinux '
>    config\_args='\-des \-Doptimize=\-O2 \-g \-pipe \-Wall \-Wp\,\-D\_FORTIFY\_SOURCE=2 \-fexceptions \-fstack\-protector \-\-param=ssp\-buffer\-size=4 \-m64 \-mtune=generic \-Dversion=5\.8\.8 \-Dmyhostname=localhost \-Dperladmin=root@&#8203;localhost \-Dcc=gcc \-Dcf\_by=Red Hat\, Inc\. \-Dinstallprefix=/usr \-Dprefix=/usr \-Dlibpth=/usr/local/lib64 /lib64 /usr/lib64 \-Dprivlib=/usr/lib/perl5/5\.8\.8 \-Dsitelib=/usr/lib/perl5/site\_perl/5\.8\.8 \-Dvendorlib=/usr/lib/perl5/vendor\_perl/5\.8\.8 \-Darchlib=/usr/lib64/perl5/5\.8\.8/x86\_64\-linux\-thread\-multi \-Dsitearch=/usr/lib64/perl5/site\_perl/5\.8\.8/x86\_64\-linux\-thread\-multi \-Dvendorarch=/usr/lib64/perl5/vendor\_perl/5\.8\.8/x86\_64\-linux\-thread\-multi \-Darchname=x86\_64\-linux\-thread\-multi \-Dvendorprefix=/usr \-Dsiteprefix=/usr \-Duseshrplib \-Dusethreads \-Duseithreads \-Duselargefiles \-Dd\_dosuid \-Dd\_semctl\_semun \-Di\_db \-Ui\_ndbm \-Di\_gdbm \-Di\_shadow \-Di\_syslog \-Dman3ext=3pm \-Duseperlio \-Dinstallusrbinperl=n \-Ubincompat5005 \-Uversiononly \-Dpager=/usr/bin/less \-isr \-Dd\_gethostent\_r\_proto \-Ud\_end
> hostent\_r\_proto \-Ud\_sethostent\_r\_proto \-Ud\_endprotoent\_r\_proto \-Ud\_setprotoent\_r\_proto \-Ud\_endservent\_r\_proto \-Ud\_setservent\_r\_proto \-Dinc\_version\_list=5\.8\.7 5\.8\.6 5\.8\.5 \-Dscriptdir=/usr/bin'
>    hint=recommended\, useposix=true\, d\_sigaction=define
>    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
>    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>    use64bitint=define use64bitall=define uselongdouble=undef
>    usemymalloc=n\, bincompat5005=undef
>  Compiler&#8203;:
>    cc='gcc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-I/usr/include/gdbm'\,
>    optimize='\-O2 \-g \-pipe \-Wall \-Wp\,\-D\_FORTIFY\_SOURCE=2 \-fexceptions \-fstack\-protector \-\-param=ssp\-buffer\-size=4 \-m64 \-mtune=generic'\,
>    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fno\-strict\-aliasing \-pipe \-Wdeclaration\-after\-statement \-I/usr/local/include \-I/usr/include/gdbm'
>    ccversion=''\, gccversion='4\.1\.2 20080704 \(Red Hat 4\.1\.2\-48\)'\, gccosandvers=''
>    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
>    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
>    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
>    alignbytes=8\, prototype=define
>  Linker and Libraries&#8203;:
>    ld='gcc'\, ldflags =''
>    libpth=/usr/local/lib64 /lib64 /usr/lib64
>    libs=\-lresolv \-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>    perllibs=\-lresolv \-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>    libc=\, so=so\, useshrplib=true\, libperl=libperl\.so
>    gnulibc\_version='2\.5'
>  Dynamic Linking&#8203;:
>    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E \-Wl\,\-rpath\,/usr/lib64/perl5/5\.8\.8/x86\_64\-linux\-thread\-multi/CORE'
>    cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-g \-pipe \-Wall \-Wp\,\-D\_FORTIFY\_SOURCE=2 \-fexceptions \-fstack\-protector \-\-param=ssp\-buffer\-size=4 \-m64 \-mtune=generic'
> 
> Locally applied patches&#8203;:
> 
> 
> \-\-\-
> @&#8203;INC for perl v5\.8\.8&#8203;:
>    /usr/lib64/perl5/site\_perl/5\.8\.8/x86\_64\-linux\-thread\-multi
>    /usr/lib/perl5/site\_perl/5\.8\.8
>    /usr/lib/perl5/site\_perl
>    /usr/lib64/perl5/vendor\_perl/5\.8\.8/x86\_64\-linux\-thread\-multi
>    /usr/lib/perl5/vendor\_perl/5\.8\.8
>    /usr/lib/perl5/vendor\_perl
>    /usr/lib64/perl5/5\.8\.8/x86\_64\-linux\-thread\-multi
>    /usr/lib/perl5/5\.8\.8
>    \.
> 
> \-\-\-
> Environment for perl v5\.8\.8&#8203;:
>    HOME=/home/jpl
>    LANG=C
>    LANGUAGE \(unset\)
>    LD\_LIBRARY\_PATH \(unset\)
>    LOGDIR \(unset\)
>    PATH=/home/jpl/x86\_64bin&#8203;:/home/jpl/bin&#8203;:/bin&#8203;:/usr/bin&#8203;:/usr/local/bin&#8203;:/etc&#8203;:/usr/sbin&#8203;:/usr/etc&#8203;:/usr/common/bin&#8203;:/usr/games&#8203;:\.
>    PERL\_BADLANG \(unset\)
>    SHELL=/bin/ksh
>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Jul 27\, 2010 at 12&#8203;:27&#8203;:31PM \-0700\, David E\. Wheeler wrote&#8203;:
> Time to switch pod2html to use Pod&#8203;::Simple&#8203;::HTMLLegacy?
> 
>   http&#8203;://www\.mail\-archive\.com/pod\-people@&#8203;perl\.org/msg01326\.html

"Works on my machine"&#8203;:
&nbsp;   http&#8203;://www\.mail\-archive\.com/pod\-people@&#8203;perl\.org/msg01325\.html

\(in that thread\. To quote it here\.\.\.\)

&nbsp;    I would love to see Pod&#8203;::Html \(and Pod&#8203;::Checker\) replaced with Pod&#8203;::Simple
&nbsp;    based equivalents\. But I don't have the time to do this\.

&nbsp;    http&#8203;://news\.perlfoundation\.org/2010/07/2010q3\-call\-for\-grant\-proposal\.html
&nbsp;    might be useful for others\, if time == money \(or they can define an
&nbsp;    exchange rate between the two\)

Nicholas Clark

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Jul 27 07&#8203;:32&#8203;:19 2010\, jpl@&#8203;research\.att\.com wrote&#8203;:
> File t\.pod contains
> 
>     =head2 title
> 
>     B\<\< foo\->\{bar\} >>
> 
> If I run
> 
>     /honk/perl5\.12\.1/bin/pod2html t\.pod > t\.html
> 
> \(to get the 5\.12\.1 version\)\, the relevant part of t\.html
> looks like
> 
>     \<h2>\<a name="title">title\</a>\</h2>
>     \<p>\<strong>foo\-\</strong>\{bar\} &gt;&gt;\</p>

This was fixed in Marc Green’s recent Pod&#8203;::Html overhaul that was
integrated into blead with commit cfe287a06\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
